### PR TITLE
Tests: ListMultipart tests

### PR DESCRIPTION
**Pull request by hackintoshrao**

### CODE REVIEW

File: object-api-multipart_test.go [hunk 1]
```diff
@@ -219,3 +219,491 @@ func testObjectAPIPutObjectPart(obj ObjectLayer, instanceType string, t *testing
 		}
 	}
 }
+
+// Wrapper for calling ListMultipartUploads tests for both XL muliple disks and single node setup.
+// TestListMultipartUploads - Tests validate listing of multipart uploads.
+func TestListMultipartUploads(t *testing.T) {
+	// getSingleNodeObjectLayer - Instantiates single node object layer and retuns it.
+	initNSLock()
```
Comment (harshavardhana) [inline object-api-multipart_test.go]: Move the initNSLock inside the closure. Do you see a reason to keep it outside?

File: object-api-multipart_test.go [hunk 2]
```diff
@@ -219,3 +219,491 @@ func testObjectAPIPutObjectPart(obj ObjectLayer, instanceType string, t *testing
...
+	// Bucket to test for mutiple upload Id's for a given object.
+	err = obj.MakeBucket(bucketNames[1])
+	if err != nil {
+		// Failed to create newbucket, abort.
+		t.Fatalf("%s ", err.Error())
+	}
+	for i := 0; i < 3; i++ {
+		// Initiate Multipart Upload on bucketNames[1] for the same object 3 times.
+		//  Used to test the listing for the case of multiple uploadID's for a given object.
+		uploadID, err := obj.NewMultipartUpload(bucketNames[1], objectNames[0], nil)
```
Comment (harshavardhana) [inline object-api-multipart_test.go]: Shadowed 'err' here - needs a fix.

File: object-api-multipart_test.go [hunk 3]
```diff
@@ -219,3 +219,491 @@ func testObjectAPIPutObjectPart(obj ObjectLayer, instanceType string, t *testing
...
+		// Create parts with 3 uploadID's for the same object.
+		// Testing for listing of all the uploadID's for given object.
+		// Insertion with 3 different uploadID's are done for same bucket and object.
+		{bucketNames[1], objectNames[0], uploadIDs[1], 1, "abcd", "e2fc714c4727ee9395f324cd2e7f331f", int64(len("abcd")), "e2fc714c4727ee9395f324cd2e7f331f"},
+		{bucketNames[1], objectNames[0], uploadIDs[2], 1, "abcd", "e2fc714c4727ee9395f324cd2e7f331f", int64(len("abcd")), "e2fc714c4727ee9395f324cd2e7f331f"},
+		{bucketNames[1], objectNames[0], uploadIDs[3], 1, "abcd", "e2fc714c4727ee9395f324cd2e7f331f", int64(len("abcd")), "e2fc714c4727ee9395f324cd2e7f331f"},
+	}
+	// Iterating over creatPartCases to generate parts.
+	for _, testCase := range createPartCases {
+		_, err := obj.PutObjectPart(testCase.bucketName, testCase.objName, testCase.uploadID, testCase.PartID, testCase.intputDataSize,
```
Comment (harshavardhana) [inline object-api-multipart_test.go:332]: Shadowing again, same as above.

File: object-api-multipart_test.go (hunk 1 above)
Comment (hackintoshrao) [inline object-api-multipart_test.go]: Just to make it more visible.

**Comment by harshavardhana** [comment]
Patch is merged as well along with @krishnasrinivas changes.

**Pull request closed without merging**